From: Somnath Kotur <somnath.kotur@emulex.com>
Date: Wed, 2 May 2012 03:40:16 +0000
Subject: [PATCH 54/58] be2net: Fix to not set link speed for disabled
 functions of a UMC card

commit 22ca7a6e9a68a7e0c149b22ebed9dcd106245bb7 upstream.

This renders the interface view somewhat inconsistent from the Host OS POV
considering the rest of the interfaces are showing their respective speeds
based on the bandwidth assigned to them.

Signed-off-by: Somnath Kotur <somnath.kotur@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/ethernet/emulex/benet/be_ethtool.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-3.2.46/drivers/net/ethernet/emulex/benet/be_ethtool.c
===================================================================
--- linux-3.2.46.orig/drivers/net/ethernet/emulex/benet/be_ethtool.c	2013-07-26 19:22:01.000000000 +0000
+++ linux-3.2.46/drivers/net/ethernet/emulex/benet/be_ethtool.c	2013-07-26 19:22:03.000000000 +0000
@@ -558,7 +558,7 @@
 				be_link_status_update(adapter, link_status);
 			if (link_speed)
 				et_speed = link_speed * 10;
-			else
+			else if (link_status)
 				et_speed = convert_to_et_speed(port_speed);
 		} else {
 			et_speed = adapter->phy.forced_port_speed;
